src/System.Net.Primitives/src/System/Net/NetworkCredential.cs

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+using System.Threading;
+using Microsoft.Win32;
+
+namespace System.Net
+{
+    /// <devdoc>
+    ///    <para>
+    ///       Provides credentials for password-based authentication schemes such as basic,
+    ///       digest, NTLM and Kerberos.
+    ///    </para>
+    /// </devdoc>
+    public class NetworkCredential : ICredentials, ICredentialsByHost
+    {
+        private static readonly object s_lockingObject = new object();
+        private string _domain;
+        private string _userName;
+        private SecureString _password;
+
+        public NetworkCredential()
+            : this(string.Empty, string.Empty, string.Empty)
+        {
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Initializes a new instance of the <see cref='System.Net.NetworkCredential'/>
+        ///       class with name and password set as specified.
+        ///    </para>
+        /// </devdoc>
+        public NetworkCredential(string userName, string password)
+        : this(userName, password, string.Empty)
+        {
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Initializes a new instance of the <see cref='System.Net.NetworkCredential'/>
+        ///       class with name and password set as specified.
+        ///    </para>
+        /// </devdoc>
+        public NetworkCredential(string userName, SecureString password)
```

Comment:
@weshaggard, SecureString is showing up here in public surface area.  Is this not part of the contract?